src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,462 @@
+import { withRouter, RouteComponentProps } from 'react-router-dom';
+
+import React, { Component } from 'react';
+import {
+    Button,
+    Modal,
+    Form,
+    Checkbox,
+    Table,
+    Input,
+    Select,
+    Label,
+    Dropdown,
+} from 'semantic-ui-react';
+
+interface IGProfilerPopupState {
+    error: string;
+    showModal: boolean;
+    topNumFeatures: number[];
+    availableOrganisms: object[];
+    selectedOrganism: string;
+    selectedSortBy: string;
+    gProfilerToken: string;
+}
+
+interface IGProfilerPopupProps {
+    featureMetadata: any;
+    availableSortBy: object[];
+}
+
+const INITIAL_STATE = {
+    error: null,
+    showModal: false,
+    topNumFeatures: [],
+    availableOrganisms: [],
+    selectedOrganism: null,
+    selectedSortBy: null,
+    gProfilerToken: null,
+};
+
+class GProfilerPopup extends Component<
+    IGProfilerPopupProps & RouteComponentProps,
+    IGProfilerPopupState
+> {
+    featureMetadata: any;
+    featureMetricTable: any;
+    availableSortBy: object[];
+
+    constructor(props: IGProfilerPopupProps & RouteComponentProps) {
+        super(props);
+        this.state = INITIAL_STATE;
+        this.featureMetadata = props.featureMetadata;
+        this.availableSortBy = props.featureMetadata.metrics.map(
+            (metric, idx) => {
+                return {
+                    key: idx,
+                    text: metric.name,
+                    value: metric.accessor,
+                };
+            }
+        );
+        this.featureMetricTable = this.buildMetricTable();
+        this.handleOpenModal.bind(this);
```

Comment:
Are all of these binds really necessary? What is the `this` context?

Reply:
Yes unfortunately, `this` is the component class context